igcollect/logfile_values.py

```diff
@@ -125,19 +125,33 @@ def get_speed(self):
         # Speed :-/?
         return self.get_sum() / self.get_timeshift()
 
+    def get_distribution(self):
+        d = {}
+        uniq_values = set(self.values)
+        for v in uniq_values:
+            d[int(v)] = self.values.count(v)
+        return d
+
     def get_metric_value(self):
-        if self.values:
-            if self.function:
-                if 'count_' in self.function:
-                    if 'percentage' in self.function:
-                        return float(
-                            self.get_count_percentage(
-                                int(self.function.split('_')[1])))
-                    return float(
-                        self.get_count(int(self.function.split('_')[1])))
-                return float(getattr(self, 'get_' + self.function)())
+
+        if not self.values:
+            return 0
+
+        if not self.function:
             return float(self.get_last_value())
-        return 0
+
+        if 'distribution' in self.function:
+            return getattr(self, 'get_' + self.function)()
+
+        if 'count_' in self.function:
```

Comment:
Is `in` really correct, or should it be rather `self.function.startswith('count_')`? Right now, this would also be triggered when `self.function` is something like `someother_count_something`.